### DISCUSSION TITLE
BME280 integration

### DISCUSSION THREAD

**Discussion by 1coderookie** [fredlcore/BSB-LAN#220] [Ideas]
Hi guys,
BSB-LAN already supports DS18B20 (temp.) and DHT22 (temp.&hum.) sensors. If it doesn't take too much memory, it could be nice for some users to have the possibility to also have support for the BME280 sensor type within BSB-LAN. It offers temperature, humidity and pressure. The BME280 is said to be more precise then the DHT22, works with the 3,3V the Due offers and will be connected to the I2C bus. 
Cheers

**Comment by 1coderookie** [comment]
I'm just mentioning @fredlcore and @dukess explicitly now to make sure you guys will see this cuz I didn't posted it within the 'Issues'.. ;)

Btw: Iirc, the BME280 could also be connected at the SPI..

**Comment by dukess** [comment]
Hi! I found some libraries. Add connection with BME280 (up to two) looks simple. But need testers because i have no hardware. As usually.

**Reply by 1coderookie** [reply]
I have BME280 here, so testing shouldn't be a problem ;)
But I'm not sure about I2C and the already existing hardware like the eeprom and the already used sda/scl-pins for the adapter. Looking at this page https://www.arduino.cc/en/Reference/Wire then we also have a second I2C-connection at SDA1 and SCL1 for an optional BME. 
Here https://forum.arduino.cc/index.php?topic=503482.msg3436090#msg3436090 I found an example of how they connected a BMP180 (it's not the same like BME280 when it comes down to functionality though, BME280 offers more) and an EEPROM. So maybe we can learn from that..? 
Maybe @fredlcore knows more about it..?

**Reply by fredlcore** [reply]
I'm not really well informed about both I2C as well as SPI. I just know that SPI is not really an option if we want to keep open the option for WiFi (which certainly more people will use compared to BME sensors). If I2C can be used with both the EEPROM as well as the sensor, then this would be a feasible option, though.

**Comment by fredlcore** [comment]
Hi @1coderookie, please open up an issue for this, so we can keep better track of it. Thanks!
As for SPI, it would be better not to use that approach because I just found a WiFi library that uses a NodeMCU as a WiFi interface. Not via serial (as the previous approach), but via SPI. The thing is that users would have to wire this "on the fly" as no ready shields are available. The same goes for these sensors, so both would be mutually exclusive. Therefore, it would be better to connect these sensors via ordinary pins if possible.

**Reply by 1coderookie** [reply]
Oh, didn't saw your post right now when I answered Sergey.. Ok, will open an Issue ;)

**Comment by 1coderookie** [comment]
Issue #235 opened

**Comment by fredlcore** [comment]
Implemented with https://github.com/fredlcore/BSB-LAN/pull/311
